client/go/outline/iptable/stream_dialer.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2025 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iptable
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/netip"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+func lookupInTable[D any](table IPTable[D], address string) (foundDialer D, ok bool) {
```

Comment:
Since this is shared, perhaps move it to a place not specific to the stream dialer. Maybe it should be part of `iptable.go`.